config/config.go

```diff
@@ -446,6 +451,8 @@ type GlobalConfig struct {
 	// Keep no more than this many dropped targets per job.
 	// 0 means no limit.
 	KeepDroppedTargets uint `yaml:"keep_dropped_targets,omitempty"`
+	// Allow UTF8 Metric and Label Names
+	MetricNameValidationScheme string `yaml:"metric_name_validation_scheme,omitempty"`
```

Comment:
Why do we have two places to configure this? We are checking that setting this to utf8 requires setting the feature flag as well. So why not just have the flag? It may make sense to introduce this config when we don't have the feature flag anymore. Or is it planned to be not a feature flag in 3.0?

Reply:
https://github.com/prometheus/prometheus/pull/14490#discussion_r1718942757

Reply:
The thinking was that deployments could have a weird setup where a certain scrape config needs to be treated differently than the rest.  for instance, for some reason they want to preserve the escaping scheme to keep metric names consistent so that alerts or dashboards don't break.

Admittedly this could be YAGNI -- I don't know all of the configurations people have and what is realistic. It could be that it's much more likely that "all or nothing" is good enough.

But also, I was having a problem in implementation that the individual scraping configs needed the term, and the global config *also* needed the term, because at the point we are checking the value, the code only has access to the individual config. I would be ok with having a single global config but then I'd need a little guidance in how to plumb that value down to the implementation.

Reply:
I think the current one makes sense now. It will be useful to do a progressive rollout of utf8 - it's normal to have one Prometheus scraping stuff for variety of targets with different owners.